docs/notebooks/asynchronous_batch_bo.pct.py

```diff
@@ -0,0 +1,197 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.12.0
+#   kernelspec:
+#     display_name: 'Python 3.7.5 64-bit (''.venv'': venv)'
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Asynchronous batch Bayesian optimization
+#
+# As shown in [Asynchronous Bayesian Optimization](asynchronous_bayesian_optimization.ipynb) tutorial, Trieste provides support for running observations asynchronously. In that tutorial we used greedy batch acquisition function called Local Penalization, and collected one new point whenever an observation was received. We also used Python multiprocessing module to run distributed observations in parallel.
+#
+# Here we demonstrate slightly different way of doing asynchronous BO. First, we make use of a non-greedy batch acquisition function, known as Batch Monte Carlo Expected Improvement. Second, we wait for several workers to finish, and then launch a new batch of points. However, since our batch size is smaller than the number of workers available, this approach is a hybrind between completely asynchronous and completely synchronous batch optimization. Third, we use [Ray](https://www.ray.io/) for parallel processing, to hide away most of the complexity of managing distributed workloads.
```

Comment:
Here**,** we demonstrate **a** slightly different way of doing asynchronous BO. First, we make use of a non-greedy batch acquisition function, known as Batch Monte Carlo Expected Improvement. Second, we wait for several workers to finish, and then launch a new batch of points. However, since our batch size is smaller than the number of workers available, this approach is a **hybrid** between completely asynchronous and completely synchronous batch optimization. Third, we use [Ray](https://www.ray.io/) for parallel processing, to hide away most of the complexity of managing distributed workloads.